tasks/fine_tuning_sbert/notebooks/Fine_tuning_SBERT_with_callback.ipynb

```diff
@@ -1,3160 +1,2839 @@
-     "text": [
-      "Grid Search Fine tuning parameters:\n",
-      "{\n",
-      "    \"all_dev_perc\": [\n",
-      "        0.2\n",
-      "    ],\n",
-      "    \"model_names\": [\n",
-      "        \"paraphrase-xlm-r-multilingual-v1\"\n",
-      "    ],\n",
-      "    \"output_path\": \"/content/drive/MyDrive/WRI-LatinAmerica-Talent/Modeling/Model_reproducibility/Model_results/BinaryClassificationExperiments/9/\",\n",
-      "    \"experiment\": 9,\n",
-      "    \"max_num_epochs\": 10,\n",
-      "    \"baseline\": 0.001,\n",
-      "    \"patience\": 5,\n",
-      "    \"seeds\": [\n",
-      "        12\n",
-      "    ],\n",
-      "    \"eval_classifier\": \"SBERT\"\n",
-      "}\n"
-     ]
-    },
+  "cells": [
     {
-     "data": {
-      "application/vnd.jupyter.widget-view+json": {
-       "model_id": "fb9db8af9feb4181aef910c5df81ea9b",
-       "version_major": 2,
-       "version_minor": 0
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "bk0XWLfMvDXU"
       },
-      "text/plain": [
-       "HBox(children=(FloatProgress(value=0.0, max=1014059776.0), HTML(value='')))"
+      "source": [
+        "# Fine tuning with S-BERT\n",
+        "\n",
+        "The models will be saved in this folder - https://drive.google.com/drive/folders/1ui5rywdrDdbBUeHph9AaA0VpfX01bbRe?usp=sharing\n"
       ]
-     },
-     "metadata": {
-      "tags": []
-     },
-     "output_type": "display_data"
     },
     {
-     "name": "stdout",
-     "output_type": "stream",
-     "text": [
-      "\n"
-     ]
-    },
-    {
-     "data": {
-      "application/vnd.jupyter.widget-view+json": {
-       "model_id": "f48d36b401ca4c4bab3ffe4a6ef1a9f7",
-       "version_major": 2,
-       "version_minor": 0
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "tW6NlcIEvXbG"
       },
-      "text/plain": [
-       "HBox(children=(FloatProgress(value=0.0, description='Epoch', max=10.0, style=ProgressStyle(description_width='…"
+      "source": [
+        "# Setup"
       ]
-     },
-     "metadata": {
-      "tags": []
-     },
-     "output_type": "display_data"
     },
     {
-     "data": {
-      "application/vnd.jupyter.widget-view+json": {
-       "model_id": "7acab4d9476c46f9bdb420a1c44d41d9",
-       "version_major": 2,
-       "version_minor": 0
+      "cell_type": "code",
+      "metadata": {
+        "collapsed": true,
+        "id": "Ml7Gd4VgxW1d",
+        "jupyter": {
+          "outputs_hidden": true
+        },
+        "colab": {
+          "base_uri": "https://localhost:8080/"
+        },
+        "outputId": "475d7585-2b94-435c-c32c-1061872f956d"
       },
-      "text/plain": [
-       "HBox(children=(FloatProgress(value=0.0, description='Iteration', max=41.0, style=ProgressStyle(description_wid…"
+      "source": [
+        "# Install necessary libraries\n",
+        "! pip install --quiet \\\n",
+        "  scprep\\\n",
+        "  spacy==3.0.5 \\\n",
+        "  wandb \\\n",
+        "  onnx \\\n",
```

Comment:
Let's remove unused packages from here (onnx and onnxruntime), I'm not sure if we ever end up using spacy but if not, let's get rid of it.

Reply:
Ah yes! Sorry about leaving those there. Removed in the next push.

For spacy, I think we use it because the ``src`` python files need it for preprocessing the sentences. @jordiplanascuchi do you think we need spacy every time we run the notebook for fine tuning?

Reply:
Actually, I just checked this and spacy needs to be there otherwise the repo import fails in the notebook

Reply:
Actually it doesn't 😅 moved spacy installation to a sbert evaluation function